### ISSUE TITLE
iPhone; When running code multiple times, different results

### ISSUE THREAD

**Issue by pimoudeveldhuis** [niklasvh/html2canvas#1663]
Using the latest version (just updated it to check, but already had it) on a iPhone 6s (but I can imagine the problem is on more iOS devices) creating the same screenshot multiple times give different results.

On Safari (macOS) everything works great, but when I used the same page on iOS it gives a very wide image (it should be 345px wide but it is over 1000px wide). The content is in it, but unreadable because of the width of the image.

I tried creating it multiple times and after two or three passings the correct image gets appended. When doing this with a setInterval instead of poking the button multiple times it also works like this. The first two or three times it's a really wide image, the third or fourth it usually starts to be the correct image. This behavior is consistent if I refresh the page and try again.

`html2canvas($(".example").get(0), { 
                    width: parseInt($(".example").css('width'), 10), 
                    height: parseInt($(".example").css('height'), 10),
                    backgroundColor: '#4E4E4E',
                }).then(function(canvas) {
                    document.body.appendChild(canvas); 
                });`

tldr; works on Safari@macOS, but gives different results on each try on Safari@iOS.

**Comment by pimoudeveldhuis** [comment]
I just did some extra testing and it seems that the canvas does not have the dimensions that it should have. On the PC it is the expected with and height of the element, but on the phone it is consistent about 8 times the width and twice the height.
